backend/api/src/model/event.rs

```diff
@@ -0,0 +1,56 @@
+use juniper::{graphql_object, FieldResult};
+
+use crate::{Context, Id, id::Key, util::RowExt};
+
+
+pub(crate) struct Event {
+    key: Key,
+    title: String,
+    video: String,
+    description: Option<String>,
+}
+
+#[graphql_object]
+impl Event {
+    fn id(&self) -> Id {
+        Id::event(self.key)
+    }
+
+    fn title(&self) -> &str {
+        &self.title
+    }
+
+    fn video(&self) -> &str {
+        &self.video
+    }
+
+    fn description(&self) -> Option<&str> {
+        self.description.as_deref()
+    }
+}
+
+impl Event {
+    pub async fn from_context(id: Id, context: &Context) -> FieldResult<Option<Self>> {
```

Comment:
I probably just call it `new` or `load`. I can't quite put it into words but `from_context` doesn't seem quite right.

Reply:
I was looking at `Realm`, which has `from_db`; `BlockValue` calls it `fetch_for_realm`. If I now call this `new` or `load`, I think the others should change as well. Or we converge on one of the existing schemas. What do you think? :thinking:

Reply:
Yeah, those are not great. How about:

- `Event::load`: it gets an ID for an event, should be clear what it does
- `RealmTree::load`: only gets the DB, should also be perfectly clear
- `BlockValue::load_all_of_realm`: here, just `load` wouldn't work I think. `all_of_realm` is still awkward tho, but a bit clearer than `for_realm`? We can also drop the `load` in that case, if you prefer that. Or maybe you have a better idea.

Reply:
When I originally wrote `from_db` and now `from_context`, I was kinda immitating the stdlib stdlib style `from_`, `with_` and similar prefixes. Which I kind of like. :thinking: 

Now that I think about it, though, the context isn't really the important thing here, right? Maybe `by_id` would be better. But the context is also not unimportant, so maybe `from_context_by_id` ... This is getting out of hand, so yeah, maybe just `load` might be good. And then if we ever want to query more than one event, we can add `load_all` or whatever. I think I can agree to that.

But then I would really make it consistent, so `load_all_of_realm` or just `load_for_realm` or something like that. But `load` should be there to indicate that it hits the DB as opposed to just creating an entity from some given values.

Should I add a commit that unifies it all?

Reply:
So yeah, I also like the `from_` style, but only if it makes sense. The context for me is not something "application logic" but just implementation detail, so I would not mention it in the name. I like `by_id`. And if we want to use "load" as indicator for "hits the DB", we can also use `Event::load_by_id`. So yeah: `RealmTree::load`, `BlockValue::load_for_realm` and (`Event::load` or `Event::load_by_id`).